src/components/protocol_handler/src/protocol_handler_impl.cc

```diff
@@ -2083,10 +2122,20 @@ void ProtocolHandlerImpl::NotifySessionStarted(
         protocol_version,
         packet->service_type(),
         rejected_params,
-        "Certain parameters in the StartService request were rejected");
+        "Certain parameters in the StartService request were rejected",
+        full_version);
+    context.is_start_session_failed_ = true;
   }
 }
 
+void ProtocolHandlerImpl::NotifySessionStarted(
+    const SessionContext& context,
+    std::vector<std::string>& rejected_params,
+    const std::string err_reason) {
+  NotifySessionStarted(
+      const_cast<SessionContext&>(context), rejected_params, err_reason);
```

Comment:
I fear this will cause undefined behavior if called with a const context.

We need to preserve the old functionality (of calling with a const context) until a major version bump.

Reply:
@iCollin agree, it's unsafe code. I replaced this cast with the local copy of the same context. As const version of deprecated function is not supposed to change the context, but supposed to use the same callbacks, I believe we can work in this way - https://github.com/smartdevicelink/sdl_core/pull/3647/commits/58fb660c930d0fe7cd42cd8123e12c4522363dc2
Note, that deprecated function is unused for now.